Makefile

```diff
@@ -0,0 +1,39 @@
+# several pytest settings
+DURATION ?= all  # pytest duration, one of short, long or all
+WORLD_SIZE ?= 1  # world size for DDP tests
+EXTRA_ARGS ?=  # additional arguments
+
+EXTRA_ARGS := --duration $(DURATION) $(EXTRA_ARGS)
+
+dirs := composer examples tests
+
+# run this to autoformat your code
+style:
+	isort -i $(dirs)
+	yapf -ri $(dirs)
+	docformatter -ri --wrap-summaries 120 --wrap-descriptions 120 $(dirs)
+
+# this only checks for style & pyright, makes no code changes
+lint:
+	isort $(dirs) -c
+	yapf -dr $(dirs)
+	docformatter -r --wrap-summaries 120 --wrap-descriptions 120 $(dirs)
+	pyright $(dirs)
+
+test:
+	pytest tests/ $(EXTRA_ARGS)
+
+test-gpu:
```

Comment:
We will want to run the gpu and deepspeed tests with and without the launcher script, since some of the gpu and deepspeed tests have world sizes of 2. I'm wondering whether it would make sense to have gpu or deepspeed be a flag like we have with world_size?

Reply:
Would it be possible for the command that jenkins runs to be something (or somethings) in the makefile? Would prefer not to hide that we run distributed GPU tests like this in the jenkinsfile.